### PR TITLE
fix(fetch.signal) capture socket and shutdown on abort signal

### PR DESCRIPTION
**Pull request by cirospaciari**

### CODE REVIEW

File: src/http_client_async.zig [hunk 1]
```diff
@@ -552,6 +557,18 @@ pub const HTTPThread = struct {
     }
 
     fn drainEvents(this: *@This()) void {
+        while (this.queued_shutdowns.pop()) |http| {
+            if (socket_async_http_abort_tracker.get(http.async_http_id)) |socket_ptr| {
```
Comment (Jarred-Sumner) [inline src/http_client_async.zig]: one of the functions lets you get and delete in the same call

I think its

fetchSwapRemove

Comment (cirospaciari) [reply src/http_client_async.zig]: you are right, just changed in https://github.com/oven-sh/bun/pull/2143/commits/097ac1ff65c29b2596a626c42ca4ba472558edee

File: src/bun.js/webcore/response.zig [hunk 2]
```diff
@@ -626,6 +628,9 @@ pub const Fetch = struct {
         signal: ?*JSC.AbortSignal = null,
         aborted: std.atomic.Atomic(bool) = std.atomic.Atomic(bool).init(false),
 
+        //needed to shutdown when aborted
+        socket: ?*uws.Socket = null,
```
Comment (Jarred-Sumner) [inline src/bun.js/webcore/response.zig]: do we still need this?

File: src/http_client_async.zig [hunk 3]
```diff
@@ -1000,23 +1026,10 @@ proxy_authorization: ?[]u8 = null,
...
-    return HTTPClient{
-        .allocator = allocator,
-        .method = method,
-        .url = url,
-        .header_entries = header_entries,
-        .header_buf = header_buf,
-        .aborted = signal,
-    };
+pub fn init(allocator: std.mem.Allocator, method: Method, url: URL, header_entries: Headers.Entries, header_buf: string, signal: ?*std.atomic.Atomic(bool)) HTTPClient {
+    return HTTPClient{ .allocator = allocator, .method = method, .url = url, .header_entries = header_entries, .header_buf = header_buf, .aborted = signal };
```
Comment (Jarred-Sumner) [inline src/http_client_async.zig]: if you use `,` at the end, it will auto format more pretty

File: src/bun.js/webcore/response.zig (hunk 2 above)
Comment (cirospaciari) [reply src/bun.js/webcore/response.zig]: i forgot this unused, removed on https://github.com/oven-sh/bun/pull/2143/commits/77368b63a8bca0fa27d529556d156198aea87a38

File: src/http_client_async.zig (hunk 3 above)
Comment (cirospaciari) [reply src/http_client_async.zig]: the comma at the end did the trick thanks, fixed in https://github.com/oven-sh/bun/pull/2143/commits/77368b63a8bca0fa27d529556d156198aea87a38

File: src/http_client_async.zig [hunk 4]
```diff
@@ -39,6 +39,9 @@ const Batch = NetworkThread.Batch;
 const TaggedPointerUnion = @import("./tagged_pointer.zig").TaggedPointerUnion;
 const DeadSocket = opaque {};
 var dead_socket = @intToPtr(*DeadSocket, 1);
+//TODO: this needs to be freed when Worker Threads are implemented
+var socket_async_http_abort_tracker = std.AutoArrayHashMap(u32, *uws.Socket).init(bun.default_allocator);
+var async_http_id: std.atomic.Atomic(u32) = std.atomic.Atomic(u32).init(0);
```
Comment (Jarred-Sumner) [inline src/http_client_async.zig:44]: what happens on overflow? does it wrap?

Comment (cirospaciari) [reply src/http_client_async.zig:44]: wraps the value

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**